tutorials/radiant-mlhub-on-demand-training-data.ipynb

```diff
@@ -0,0 +1,1281 @@
+    "]\n",
+    "len(label_item_links)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b9836f40-4228-4d72-8b84-d94cae1030c6",
+   "metadata": {},
+   "source": [
+    "Now we take a random sample that is 1/100th the original dataset size"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "97b8f303-08c8-4353-847e-fc4d712edac6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "label_item_sample = np.random.choice(\n",
+    "    a=label_item_links, size=int(len(label_item_links) / 100), replace=False\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8c171723-8bff-4755-aa56-2db7d2074011",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "label_item_sample[0:5]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bf4e290e-f7e4-4906-a44c-0435f6ce8f3b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rand_idx = np.random.randint(len(label_item_sample))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a05b5914-ddf8-4332-87f9-4e631e71110a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "first_label_item = Item.from_file(\n",
+    "    os.path.join(TMP_DIR, BIGEARTHNET_LABEL_COLLECTION, label_item_sample[rand_idx])\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "65959f74-4d1d-4ce4-8583-bc51d6047ce1",
+   "metadata": {},
+   "source": [
+    "Chip ID for the sample label Item pulled:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4420d40b-2cd3-4f74-b271-a137ca71e360",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "first_label_item.id"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c0b0734b-ce42-48de-8bea-f4c4f5e9b37f",
+   "metadata": {},
+   "source": [
+    "Links for the sample label Item, take special note of the `rel=source` Link listed:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "42d94501-6305-4329-aa4f-ef69802951b8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "first_label_item.links"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5d23fb83-400e-487b-940f-fc158b5ae41b",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "### Fetch source items for random sample from BigEarthNet"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e419d96f-f8b9-4911-b5d1-4a4077767062",
+   "metadata": {},
+   "source": [
+    "If we had the source collection archive downloaded and uncompressed in the same parent directory as the labels collection, we could reference the source Items and images directly. However the BigEarthNet source collection is over 60GB when compressed. Therefore to work around the disk size limitations of a Planetary Computer instance, we can query the same source items from the MLHub API endpoint, the same way we got the labels, but filter to the exact source item using IDs."
```

Comment:
What's your preferred workflow as a user, and from MLHub's point of view? Do you want people making local copies of this dataset, or do you want them fetching from your storage on demand?

IMO, the ideal workflow is on-demand fetching from blob storage in the same region + caching, but I'm curious what you think.